src/msg/async/AsyncConnection.cc

```diff
@@ -2299,15 +2299,20 @@ void AsyncConnection::handle_ack(uint64_t seq)
 {
   ldout(async_msgr->cct, 15) << __func__ << " got ack seq " << seq << dendl;
   // trim sent list
-  std::lock_guard<std::mutex> l(write_lock);
+  std::vector<Message*> pending;
```

Comment:
small_vector here?

Reply:
boost/container/small_vector.hpp?

Reply:
yeah.  it'll avoid an allocation in the general case